include/motor-control/core/motor_messages.hpp

```diff
@@ -17,23 +17,34 @@ struct MotionConstraints {
     um_per_tick_sq max_acceleration;
 };
 
-struct Move {
+enum class MoveStopCondition : uint8_t {
+    none = 0x0,
+    limit_switch = 0x1,
+    cap_sensor = 0x2
+};
+
+struct Move {        // NOLINT(cppcoreguidelines-pro-type-member-init)
     ticks duration;  // in ticks
     steps_per_tick velocity;
     steps_per_tick_sq acceleration;
     uint8_t group_id;
     uint8_t seq_id;
+    MoveStopCondition stop_condition = MoveStopCondition::none;
 };
-
 const uint8_t NO_GROUP = 0xff;
 
-enum class AckMessageId : uint8_t { complete = 0x1, error = 0x04 };
+enum class AckMessageId : uint8_t {
+    complete = 0x1,
+    triggered_lim_sw = 0x2,
+    error = 0x04
+};
 
 struct Ack {
     uint8_t group_id;
     uint8_t seq_id;
     uint32_t current_position;
     AckMessageId ack_id;
+    bool lim_sw_triggered;
```

Comment:
I also think the `bool lim_sw_triggered` is redundant given that we have the `AckMessageId` already, I don't believe we need it